dashboard/lib/school_info_deduplicator.rb

```diff
@@ -0,0 +1,45 @@
+module SchoolInfoDeduplicator
+  # Returns true if the SchoolInfo already exists and we should reuse that.
+  # Returns false if the SchoolInfo is new and should be stored.
+  # Validates the SchoolInfo first so that we fall into the latter path in
+  # that case.
+  def deduplicate_school_info(school_info_attr, self_object)
```

Comment:
This actually does apply the de-duping as well, as a side-effect - notice that the conditional has a single equals, so does an assignment, and then we assign the self_object field if the existing is a dupe.

That said, this could probably use some cleanup, the method is kind of doing double-duty right now.